src/test/java/hello/HomePageTest.java

```diff
@@ -0,0 +1,63 @@
+package hello;
+
+// import static org.hamcrest.Matchers.equalTo;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.xpath;
+
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+public class HomePageTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Test
+    public void getHomePage_IsOk() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.get("/").accept(MediaType.TEXT_HTML))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType("text/html;charset=UTF-8"));
+    }
+
+    @Test
+    public void getHomePage_ContentType() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.get("/").accept(MediaType.TEXT_HTML))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType("text/html;charset=UTF-8"));
+    }
+
+
+    @Test
+    public void getHomePage_BootstrapLoaded() throws Exception {
+        mvc.perform(MockMvcRequestBuilders.get("/").accept(MediaType.TEXT_HTML))
+                .andExpect(status().isOk())
+                .andExpect(xpath(BootstrapLiterals.bootstrapCSSXpath).exists());
+        for (String s: BootstrapLiterals.bootstrapJSurls) {
+            String jsXPath = String.format("//script[@src='%s']",s);
+            mvc.perform(MockMvcRequestBuilders.get("/greeting").accept(MediaType.TEXT_HTML))
+              .andExpect(status().isOk())
+              .andExpect(xpath(jsXPath).exists());
+        }
+    }
+
+
+    @Test
+    public void getHomePage_hasCorrectTitle() throws Exception {
```

Comment:
It seems that these tests are boilerplate that has to go on every single page that we include javascript and bootstrap. That being said, I'm not away of any way to dry this up in JUnit given that it relies on one function = one test.

This comment is more observation than review.

Reply:
My intent in the next iteration is to factor out this code into something included on every page.  I'm not expecting that we'd have these tests for every page as the app grows, but I wanted some way to be sure that as I refactor, I don't screw things up and have the Boostrap loaded on one page but not another.

We might be able to refactor this into a simple "assertBootStrapIsLoaded(page)" call once we understand better how the testing methods work in Spring.

Thanks for the feedback